tests/abstract-tensor-tests.lisp

```diff
@@ -4,6 +4,81 @@
 
 (in-package #:magicl-tests)
 
+(defmacro swapping-arguments-is ((predicate arg1 arg2))
+  "Try both argument orders for a commutative predicate."
+  (let ((arg1sym (gensym))
+        (arg2sym (gensym)))
+    `(let ((,arg1sym ,arg1)
+           (,arg2sym ,arg2))
+       (is (,predicate ,arg1sym ,arg2sym))
+       (is (,predicate ,arg2sym ,arg1sym)))))
+
+(defmacro swapping-arguments-not ((predicate arg1 arg2))
```

Comment:
if `swapping-arguments-is` is made a function, you can use `complement` on `predicate`